pkg/image/apis/image/helper.go

```diff
@@ -1240,3 +1241,59 @@ func ValidateRegistryURL(registryURL string) error {
 	}
 	return nil
 }
+
+// UpdateWithRegistryHostnames takes ImageStreams, ImageStreamList, Image or
+// ImageList object and mutates the image pull specs using configured hostnames
+// for image registry.
+func UpdateWithRegistryHostnames(r RegistryHostnameRetriever, obj runtime.Object) error {
```

Comment:
@soltysh i added this helper because I hate this code being copied in both IS and image storages and in rest.